### PR TITLE
[FLAC] FLAC fuzzers

### PR DESCRIPTION
**Pull request by guidovranken**
https://github.com/xiph/flac/issues/125

### CODE REVIEW

**Comment by guidovranken** [comment]
This has the decoder fuzzer. Working on the encoder fuzzer.
The decoder fuzzer will find the first ASAN bug within minutes. Also noticed an MSAN bug, and an OOB write with i386 ASAN.
You can either merge now or wait until the encoder fuzzer + auto_ccs are in, whichever you prefer.

**Comment by jonathanmetzman** [comment]
> This has the decoder fuzzer. Working on the encoder fuzzer.
> The decoder fuzzer will find the first ASAN bug within minutes. Also noticed an MSAN bug, and an OOB write with i386 ASAN.
> You can either merge now or wait until the encoder fuzzer + auto_ccs are in, whichever you prefer.

I wrote an encoder fuzzer in chromium it's a little bit higher up so it fuzzes chromium bits too, but iirc underneath it is fuzzing flac. Maybe you'll find it helpful: https://cs.chromium.org/chromium/src/content/browser/speech/audio_encoder_fuzzer.cc

File: projects/flac/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+
+FROM gcr.io/oss-fuzz-base/base-builder
+MAINTAINER your@email.com
```
Comment (jonathanmetzman) [inline projects/flac/Dockerfile]: Just drop this in a future CL I've never seen a use for maintainer field.

**Comment by guidovranken** [comment]
> > This has the decoder fuzzer. Working on the encoder fuzzer.
> > The decoder fuzzer will find the first ASAN bug within minutes. Also noticed an MSAN bug, and an OOB write with i386 ASAN.
> > You can either merge now or wait until the encoder fuzzer + auto_ccs are in, whichever you prefer.
> 
> I wrote an encoder fuzzer in chromium it's a little bit higher up so it fuzzes chromium bits too, but iirc underneath it is fuzzing flac. Maybe you'll find it helpful: https://cs.chromium.org/chromium/src/content/browser/speech/audio_encoder_fuzzer.cc

Thank you. The thing is I have an encoder fuzzer but it OOM's within a minute so I'm figuring out whether I'm abusing the API or whether it's a bug in flac.

File: projects/flac/Dockerfile (hunk 1 above)
Comment (jonathanmetzman) [reply projects/flac/Dockerfile]: Did this myself

**Pull request merged**